internal/engine/eval/homoglyphs/application/mixed_scripts_eval_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package application
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	evalerrors "github.com/stacklok/minder/internal/engine/errors"
+	"github.com/stacklok/minder/internal/engine/eval/homoglyphs/domain"
+	"github.com/stacklok/minder/internal/engine/eval/templates"
+)
+
+func TestEvaluationDetailRendering(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		msg     string
+		msgArgs []any
+		tmpl    string
+		args    any
+		error   string
+		details string
+	}{
+		// mixed scripts template
+		{
+			name: "mixed scripts template with one violation",
+			msg:  "this is the message",
+			tmpl: templates.MixedScriptsTemplate,
+			args: map[string]any{
+				"violations": []*domain.Violation{
+					{
+						MixedScript: &domain.MixedScriptInfo{
+							Text:         "Бorld",
+							ScriptsFound: []string{"Cyrillic", "Latin"},
+						},
+					},
+				},
+			},
+			error:   "evaluation failure: this is the message",
+			details: "Mixed scripts found:\n* Text: `Бorld`, Scripts: [Cyrillic, Latin]",
+		},
+		{
+			name: "mixed scripts template with multiple violations",
+			msg:  "this is the message",
+			tmpl: templates.MixedScriptsTemplate,
+			args: map[string]any{
+				"violations": []*domain.Violation{
+					{
+						MixedScript: &domain.MixedScriptInfo{
+							Text:         "Бorld",
+							ScriptsFound: []string{"Cyrillic", "Latin"},
+						},
+					},
+					{
+						MixedScript: &domain.MixedScriptInfo{
+							Text:         "Ѳ.HandleFunc(\"hi\",",
+							ScriptsFound: []string{"Cyrillic", "Latin"},
+						},
+					},
+				},
```

Comment:
**nit:** what do you think about adding a test having `Text: ""` and another one having `ScriptsFound: nil` as a sort of documentation?

Reply:
I would guess that we would never be in that state because if the text or scripts aren't present, then we shouldn't have a mixed scripts violation.

Reply:
I'm going to merge this so that I can build on top of it for the other homoglyphs rule but I'm happy to continue the discussion here.